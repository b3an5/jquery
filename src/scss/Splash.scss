@font-face {
  font-family: 'chalkBoard-font';
  src: url('../fonts/font/chalkBoard.ttf')
}

.splash-page {
  background: url('../pics/chalkBoard.jpg');
  background-size: cover;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.splash-title {
  color: #fff;
  font-size: 4em;
  font-family: 'chalkBoard-font'
}

.app-description {
  border: 2px solid black;
  border-radius: 25px;
  padding: 10px;
  background-color: lightblue;
  text-align: center;
}

.splash-button {
  padding: 5px;
  color: yellow;
  background-color: green;
  margin-top: 5px;
  &:hover {
    cursor: pointer;
    color: green;
    background-color: yellow;
  }
}