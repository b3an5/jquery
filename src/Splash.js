import React from "react";
import './scss/main.scss'

export default function Splash(props) {

  return(
    <div className='splash-page'> 
      <h1 className='splash-title'>Jquery knowlege Tester</h1>
      <div className='app-description'>
      <p>Welcome to my app, this is used to test your jQuery skills.</p>
      <p>you will be presented with vanilla javascript</p>
      <p>you must write the equivilent JQuery code in the input box</p>
      </div>
      <button className='splash-button' onClick={props.showMainPage}>Click here to get started</button>
      <form className='difficulty-buttons'>
        <div className='switch'>
          <input className='radios' name='difficulty' type='radio' id='Easy' value='10' defaultChecked />
          <input className='radios' name='difficulty' type='radio' id='Medium' value='20' />
          <input className='radios' name='difficulty' type='radio' id='Hard' value='30' />
          <input className='radios' name='difficulty' type='radio' id='Very Hard' value='41' />
        </div>
      </form>
    </div>
  )
}