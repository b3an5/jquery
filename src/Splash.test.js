import React from 'react';
import Splash from './Splash.js';
import { shallow } from 'enzyme';

describe('Splash', () => {

  let wrapper;
  const mockShowMainPage = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Splash 
      showMainPage={mockShowMainPage} />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call run show main page when button is clicked', () => {
    wrapper.find('.splash-button').simulate('click')
    expect(mockShowMainPage).toBeCalled();
  })

})