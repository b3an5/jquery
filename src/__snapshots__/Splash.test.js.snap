// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    showMainPage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="splash-title"
        >
          Jquery knowlege Tester
        </h1>,
        <div
          className="app-description"
        >
          <p>
            Welcome to my app, this is used to test your jQuery skills.
          </p>
          <p>
            you will be presented with vanilla javascript
          </p>
          <p>
            you must write the equivilent JQuery code in the input box
          </p>
        </div>,
        <button
          className="splash-button"
          onClick={[MockFunction]}
        >
          Click here to get started
        </button>,
        <form
          className="difficulty-buttons"
        >
          <div
            className="switch"
          >
            <input
              className="radios"
              defaultChecked={true}
              id="Easy"
              name="difficulty"
              type="radio"
              value="10"
            />
            <input
              className="radios"
              id="Medium"
              name="difficulty"
              type="radio"
              value="20"
            />
            <input
              className="radios"
              id="Hard"
              name="difficulty"
              type="radio"
              value="30"
            />
            <input
              className="radios"
              id="Very Hard"
              name="difficulty"
              type="radio"
              value="41"
            />
          </div>
        </form>,
      ],
      "className": "splash-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jquery knowlege Tester",
          "className": "splash-title",
        },
        "ref": null,
        "rendered": "Jquery knowlege Tester",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Welcome to my app, this is used to test your jQuery skills.
            </p>,
            <p>
              you will be presented with vanilla javascript
            </p>,
            <p>
              you must write the equivilent JQuery code in the input box
            </p>,
          ],
          "className": "app-description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to my app, this is used to test your jQuery skills.",
            },
            "ref": null,
            "rendered": "Welcome to my app, this is used to test your jQuery skills.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "you will be presented with vanilla javascript",
            },
            "ref": null,
            "rendered": "you will be presented with vanilla javascript",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "you must write the equivilent JQuery code in the input box",
            },
            "ref": null,
            "rendered": "you must write the equivilent JQuery code in the input box",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click here to get started",
          "className": "splash-button",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Click here to get started",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="switch"
          >
            <input
              className="radios"
              defaultChecked={true}
              id="Easy"
              name="difficulty"
              type="radio"
              value="10"
            />
            <input
              className="radios"
              id="Medium"
              name="difficulty"
              type="radio"
              value="20"
            />
            <input
              className="radios"
              id="Hard"
              name="difficulty"
              type="radio"
              value="30"
            />
            <input
              className="radios"
              id="Very Hard"
              name="difficulty"
              type="radio"
              value="41"
            />
          </div>,
          "className": "difficulty-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="radios"
                defaultChecked={true}
                id="Easy"
                name="difficulty"
                type="radio"
                value="10"
              />,
              <input
                className="radios"
                id="Medium"
                name="difficulty"
                type="radio"
                value="20"
              />,
              <input
                className="radios"
                id="Hard"
                name="difficulty"
                type="radio"
                value="30"
              />,
              <input
                className="radios"
                id="Very Hard"
                name="difficulty"
                type="radio"
                value="41"
              />,
            ],
            "className": "switch",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radios",
                "defaultChecked": true,
                "id": "Easy",
                "name": "difficulty",
                "type": "radio",
                "value": "10",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radios",
                "id": "Medium",
                "name": "difficulty",
                "type": "radio",
                "value": "20",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radios",
                "id": "Hard",
                "name": "difficulty",
                "type": "radio",
                "value": "30",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radios",
                "id": "Very Hard",
                "name": "difficulty",
                "type": "radio",
                "value": "41",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="splash-title"
          >
            Jquery knowlege Tester
          </h1>,
          <div
            className="app-description"
          >
            <p>
              Welcome to my app, this is used to test your jQuery skills.
            </p>
            <p>
              you will be presented with vanilla javascript
            </p>
            <p>
              you must write the equivilent JQuery code in the input box
            </p>
          </div>,
          <button
            className="splash-button"
            onClick={[MockFunction]}
          >
            Click here to get started
          </button>,
          <form
            className="difficulty-buttons"
          >
            <div
              className="switch"
            >
              <input
                className="radios"
                defaultChecked={true}
                id="Easy"
                name="difficulty"
                type="radio"
                value="10"
              />
              <input
                className="radios"
                id="Medium"
                name="difficulty"
                type="radio"
                value="20"
              />
              <input
                className="radios"
                id="Hard"
                name="difficulty"
                type="radio"
                value="30"
              />
              <input
                className="radios"
                id="Very Hard"
                name="difficulty"
                type="radio"
                value="41"
              />
            </div>
          </form>,
        ],
        "className": "splash-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jquery knowlege Tester",
            "className": "splash-title",
          },
          "ref": null,
          "rendered": "Jquery knowlege Tester",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Welcome to my app, this is used to test your jQuery skills.
              </p>,
              <p>
                you will be presented with vanilla javascript
              </p>,
              <p>
                you must write the equivilent JQuery code in the input box
              </p>,
            ],
            "className": "app-description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to my app, this is used to test your jQuery skills.",
              },
              "ref": null,
              "rendered": "Welcome to my app, this is used to test your jQuery skills.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "you will be presented with vanilla javascript",
              },
              "ref": null,
              "rendered": "you will be presented with vanilla javascript",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "you must write the equivilent JQuery code in the input box",
              },
              "ref": null,
              "rendered": "you must write the equivilent JQuery code in the input box",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click here to get started",
            "className": "splash-button",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Click here to get started",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="switch"
            >
              <input
                className="radios"
                defaultChecked={true}
                id="Easy"
                name="difficulty"
                type="radio"
                value="10"
              />
              <input
                className="radios"
                id="Medium"
                name="difficulty"
                type="radio"
                value="20"
              />
              <input
                className="radios"
                id="Hard"
                name="difficulty"
                type="radio"
                value="30"
              />
              <input
                className="radios"
                id="Very Hard"
                name="difficulty"
                type="radio"
                value="41"
              />
            </div>,
            "className": "difficulty-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="radios"
                  defaultChecked={true}
                  id="Easy"
                  name="difficulty"
                  type="radio"
                  value="10"
                />,
                <input
                  className="radios"
                  id="Medium"
                  name="difficulty"
                  type="radio"
                  value="20"
                />,
                <input
                  className="radios"
                  id="Hard"
                  name="difficulty"
                  type="radio"
                  value="30"
                />,
                <input
                  className="radios"
                  id="Very Hard"
                  name="difficulty"
                  type="radio"
                  value="41"
                />,
              ],
              "className": "switch",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "radios",
                  "defaultChecked": true,
                  "id": "Easy",
                  "name": "difficulty",
                  "type": "radio",
                  "value": "10",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "radios",
                  "id": "Medium",
                  "name": "difficulty",
                  "type": "radio",
                  "value": "20",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "radios",
                  "id": "Hard",
                  "name": "difficulty",
                  "type": "radio",
                  "value": "30",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "radios",
                  "id": "Very Hard",
                  "name": "difficulty",
                  "type": "radio",
                  "value": "41",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
